spec/mongoid/association/referenced/has_and_belongs_to_many/proxy_spec.rb

```diff
@@ -3723,13 +3723,6 @@ class Distributor
     end
   end
 
-  context "when there is a foreign key in the aliased associations" do
-    it "has the correct aliases" do
-      expect(Dog.aliased_associations["breed_ids"]).to eq("breeds")
-      expect(Breed.aliased_associations["dog_ids"]).to eq("dogs")
-    end
-  end
```

Comment:
This looks like a breaking change. Aren't some applications likely to depend on this syntax?

I think we should add a feature flag here and very clearly document the change in the release notes.

@jamis please hold on merging this until above is addressed.

Reply:
`aliased_associations` hash is marked as `@api private` - [here](https://github.com/mongodb/mongoid/blob/503fba5a6cb066d9ab2c774c9922349a4cb67be5/lib/mongoid/association/macros.rb#L36C11-L36C23). So, strictly speaking, an application shouldn't depend on its internals.

Reply:
Well, if this used to work, it was definitely an unexpected side effect of MONGOID-5089. I can hardly imagine a use case where one may want to do something like this. It is the case when `Breed` class uses hashes as ids?

I'll check if something like this does work now in a reasonable situation, and document it and/or put it behind a feature flag this is the case.

Reply:
@comandeo just to be clear I am not sure the example I gave used to work, **I didn't test it**, however I do vaguely recall seeing some _ids method work in cases I wouldn't expect it to in the past (I forget exactly what.)